### PR TITLE
fix: Get the correct file path

### PR DESCRIPTION
**Pull request by eriklimakc**
GitHub Issue (If applicable): #1165

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
The url being returned is `file:\C:\C\1165\UnoApp1\UnoApp1\bin\Debug\net8.0-windows10.0.19041\AppxManifest.xml`. When `File.Exist` uses that url, it returns false due to the "file" prefix.

## What is the new behavior?
No "file" prefix.

## PR Checklist

Please check if your PR fulfills the following requirements:
- [X] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [X] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal)
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Windows.crossruntime.cs [hunk 1]
```diff
@@ -255,7 +255,10 @@ private static async Task<string> ApplicationPathFromFileName(Assembly? entry, s
 #pragma warning restore SYSLIB0044 // Type or member is obsolete
 			if (codebase is not null)
 			{
-				var manifestPath = Path.Combine(Path.GetDirectoryName(codebase) ?? string.Empty, fileName);
+				UriBuilder uri = new UriBuilder(codebase);
+				string path = Uri.UnescapeDataString(uri.Path);
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Windows.crossruntime.cs:259]: Let's add a comment here explaining the reason for this call. Mentioning the "file://' prefix that was causing problems

**Review by Xiaoy312 — APPROVED** [review]
we should probably wrap it in #If WINDOWS
given the problem is only affecting windows unpackaged
\---
or, did you checked that this doesnt break/isnt reached on skia/wasm?

**Comment by kazo0** [comment]
@eriklimakc Can you double check the comment from @Xiaoy312 and check if either the change works fine for skia/wasm or if we need to wrap this in #if WINDOWS

**Comment by eriklimakc** [comment]
> we should probably wrap it in #If WINDOWS given the problem is only affecting windows unpackaged --- or, did you checked that this doesnt break/isnt reached on skia/wasm?

**Skia**: On skia the prefix 'file:' is also present on `codebase` and the code added works fine. It didn't change the behavior, since the file on the check below doesn't exist, so it loads from `StorageFile.GetFileFromApplicationUriAsync(new Uri($"ms-appx:///{fileName}"))`.

https://github.com/unoplatform/uno.toolkit.ui/blob/43d4127894b14ec4258fe08769f006cafbdbe532/src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Windows.crossruntime.cs#L264

https://github.com/unoplatform/uno.toolkit.ui/blob/43d4127894b14ec4258fe08769f006cafbdbe532/src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Windows.crossruntime.cs#L274-L275

**Wasm**: I can't debug wasm, but I used `typeof(ExtendedSplashScreen).Log()...` to see if the code is reached, but I didn't see any of the logs I added on the Console. I didn't notice any change on the behavior.

I also downloaded the artifacts from this PR and used on a new uno app and the app didn't show any problems.

Are there other tests I could do to ensure this will run smoothly?

cc @Xiaoy312 @kazo0

**Comment by kazo0** [comment]
@eriklimakc 

I would say we should just do what @Xiaoy312 mentioned and wrap this specific change in #If WINDOWS just so we can be sure we aren't changing the logic for other platforms

**Review by kazo0 — APPROVED** [review]

**Comment by kazo0** [comment]
@mergifyio backport release/stable/6.1

**Pull request merged**